backend/typescript/services/implementations/campService.ts

```diff
@@ -26,16 +102,34 @@ class CampService implements ICampService {
       }
       const campers = camp.campers as Camper[];
 
-      return campers.map((camper) => ({
-        registrationDate: camper.registrationDate,
-        hasPaid: camper.hasPaid,
-        chargeId: camper.chargeId,
-        formResponses: camper.formResponses,
-      }));
-    } catch (error: unknown) {
-      Logger.error(
-        `Failed to get entities. Reason = ${getErrorMessage(error)}`,
+      return await Promise.all(
+        campers.map(async (camper) => {
+          const { formResponses } = camper;
+          const formResponseObject: { [key: string]: string } = {};
+          const formQuestionsPromise = Array.from(formResponses.keys()).map(
+            (questionId) => {
+              return MgFormQuestion.findById(questionId, "question");
```

Comment:
hmm I think we can get all of the formQuestions at once? to reduce individual calls to the db - we can do something like` MgFormQuestion.find({'_id': {$in: [all ids here]}})`? https://stackoverflow.com/questions/8303900/mongodb-mongoose-findmany-find-all-documents-with-ids-listed-in-array

Reply:
also is it possible to only grab all of the questions once? so outside of the campers map function, to avoid getting all the questions multiple times. you can maybe do a union of all question ids of all campers, get all of the question objects from the db (making a map of question id to question object), then for each camper create the formResponse using the map we created earlier. what are you thoughts, do you think this would be too complicated?

Reply:
thats a very good catch! I'll include it in my PR!